src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -57,7 +55,9 @@ public class BodyModelBinder : BindingSourceModelBinder
                 var unsupportedContentType = Resources.FormatUnsupportedContentType(
                     bindingContext.OperationBindingContext.HttpContext.Request.ContentType);
                 bindingContext.ModelState.AddModelError(bindingContext.ModelName, unsupportedContentType);
-                return new ModelBindingResult(null, bindingContext.ModelName, isModelSet: false);
+
+                // Base class will create a ModelBindingResult that exits current ModelBinding loop.
+                return null;
```

Comment:
as I mentioned in the PR description, it helps to look at the commit descriptions.  you're looking at part of one of the earliest commits -- the second.  in that I "reduce special-case `ModelBindingResult` creation locations (return `null`)"  to make the special-cases more visible.

unfortunately I also named parameters and improved some comments in that commit.  but if you hate the change, I can back it out or back parts of it out.